src/features/world/scenes/BaseScene.ts

```diff
@@ -9,24 +9,98 @@ import { INITIAL_BUMPKIN } from "features/game/lib/constants";
 import { BumpkinContainer } from "../containers/BumpkinContainer";
 import { interactableModalManager } from "../ui/InteractableModals";
 import {
+  ChatContext,
   ChatMessageReceived,
   MachineInterpreter,
   PlayerJoined,
   PlayerQuit,
+  RoomEvent,
+  RoomId,
 } from "../roomMachine";
 import { CONFIG } from "lib/config";
 import { NPCName, NPC_WEARABLES } from "lib/npcs";
 import { npcModalManager } from "../ui/NPCModals";
 import { BumpkinParts } from "lib/utils/tokenUriBuilder";
 import { SUNNYSIDE } from "assets/sunnyside";
+import { EventObject, State } from "xstate";
 
 export type NPCBumpkin = {
   x: number;
   y: number;
   npc: NPCName;
 };
 
-export class BaseScene extends Phaser.Scene {
+export abstract class BaseScene extends Phaser.Scene {
+  abstract roomId: RoomId;
```

Comment:
This is a reference to the current room. Inheriting classes are required to implement it.